### PR TITLE
fix: login with uppercase and mixed case role names

### LINKED ISSUE #4678: Postgrest effectively using lowercased role name sometimes
### Environment

* PostgreSQL version: 17 on debian
* PostgREST version: postgrest/postgrest:v13.0.7
* Operating system: debian 

### Description of issue

We've noticed errors related to postgrest using roles that do not exist. In our case the role postgrest is configured to use has mixed capitalization. it seems that postgrest sometimes uses a lowercase role name  (probably unintentionally by not double quoting)

relevant logs:

```
$ docker inspect --format '{{ .Config.Env }}' postgREST-production 
[
  PGRST_DB_ANON_ROLE=anon  
  PGRST_DB_PRE_REQUEST=api_authorization.set_composite_role_from_claims
  PGRST_DB_SCHEMA=api 
  PGRST_DB_URI=postgres://postgREST-service-production:xxx@192.168.1.17:5432/db 
  PGRST_JWT_SECRET=@/run/secrets/rsa.jwk 
  PGRST_LOG_LEVEL=debug
]
```

```
midgethoen@app1:~$ docker stop postgREST-production
postgREST-production
midgethoen@app1:~$ docker start -a postgREST-production 
27/Feb/2026:08:18:54 +0000: Starting PostgREST 13.0.7...
27/Feb/2026:08:18:54 +0000: Trying to borrow a connection from pool
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is being established
27/Feb/2026:08:18:54 +0000: API server listening on 0.0.0.0:3000
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is available
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is used
27/Feb/2026:08:18:54 +0000: Listening for database notifications on the "pgrst" channel
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is available
27/Feb/2026:08:18:54 +0000: Borrowed a connection from the pool
27/Feb/2026:08:18:54 +0000: Successfully connected to PostgreSQL 17.8 (Debian 17.8-0+deb13u1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 14.2.0-19) 14.2.0, 64-bit
27/Feb/2026:08:18:54 +0000: Connection Pool initialized with a maximum size of 10 connections
27/Feb/2026:08:18:54 +0000: Trying to borrow a connection from pool
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is used
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is available
27/Feb/2026:08:18:54 +0000: Borrowed a connection from the pool
27/Feb/2026:08:18:54 +0000: Failed to query database settings for the config parameters.{"code":"42704","details":null,"hint":null,"message":"role \"postgrest-service-production\" does not exist"}
27/Feb/2026:08:18:54 +0000: Trying to borrow a connection from pool
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is used
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is available
27/Feb/2026:08:18:54 +0000: Borrowed a connection from the pool
27/Feb/2026:08:18:54 +0000: Failed to query the role settings. {"code":"42704","details":null,"hint":null,"message":"role \"postgrest-service-production\" does not exist"}
27/Feb/2026:08:18:54 +0000: Config reloaded
27/Feb/2026:08:18:54 +0000: Trying to borrow a connection from pool
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is used
27/Feb/2026:08:18:54 +0000: Connection 76aa591a-5e05-4d7e-abd5-189b23671b7d is available
27/Feb/2026:08:18:54 +0000: Borrowed a connection from the pool
27/Feb/2026:08:18:54 +0000: Schema cache queried in 57.9 milliseconds
27/Feb/2026:08:18:54 +0000: Schema cache loaded 10 Relations, 50 Relationships, 12 Functions, 0 Domain Representations, 4 Media Type Handlers, 487 Timezones
27/Feb/2026:08:18:54 +0000: Schema cache loaded in 104.4 milliseconds
```

note the configured connection string and the error:
- `PGRST_DB_URI=postgres://postgREST-service-production:xxx@192.168.1.17:5432/db` 
- `Failed to query database settings for the config parameters.{"code":"42704","details":null,"hint":null,"message":"role \"postgrest-service-production\" does not exist"}
`
- `Failed to query the role settings. {"code":"42704","details":null,"hint":null,"message":"role \"postgrest-service-production\" does not exist"}`

We have recently migrated from postgres 15 -> 17. this might have caused the error to occur; or we have just now noticed it. 

postgREST seems functional despite the error. We have avoided the issue by lowercasing the role name. (the mixed capitalization is a bad habit anyway)

Can someone confirm that it is plausible that postgREST is wrongly sending a lowercases or unquoted role name or im i missing something?

### PR DESCRIPTION
**Pull request by taimoorzaeem**
PostgREST failed when querying role settings where current role name contained uppercase letters. This commit resolves it by quoting the CURRENT_USER.

Closes #4678.

### CODE REVIEW

File: src/PostgREST/Config/Database.hs [hunk 1]
```diff
@@ -168,8 +168,11 @@ queryRoleSettings pgVer prepared =
       group by kv.rolname, i.value;
     |]
 
+    -- Quote it to allow upper case or mixed case role names
+    currentUser = "quote_ident(current_user)::regrole::oid"
```
Comment (wolfgangwalther) [inline src/PostgREST/Config/Database.hs]: I'd say either move this to the top-level to use it for all 3 cases consistently - or inline it in all 3 cases. The current mix of doing it twice one way and once the other is odd :)

WDYT?

File: test/io/fixtures/roles.sql [hunk 2]
```diff
@@ -1,18 +1,19 @@
 DROP ROLE IF EXISTS
   postgrest_test_anonymous, postgrest_test_author,
   postgrest_test_serializable, postgrest_test_repeatable_read,
-  postgrest_test_w_superuser_settings;
+  postgrest_test_w_superuser_settings, "Postgrest_Test_Mixed_Case";
```
Comment (wolfgangwalther) [inline test/io/fixtures/roles.sql]: I wonder whether we need to add a new role, a new test case etc. - or whether we could just rename *an existing* authenticator role slightly, so that we're testing this case implicitly in all test-cases? We wouldn't need to add another specific test-case for that then.

It seems like we don't really need to test the inverse case, aka a fully lower-case role name, so we'd not be losing anything.

WDYT?

File: src/PostgREST/Config/Database.hs (hunk 1 above)
Comment (taimoorzaeem) [reply src/PostgREST/Config/Database.hs]: Yeah, agree. I was thinking this too. I think inlining would be better for now.

File: test/io/fixtures/roles.sql (hunk 2 above)
Comment (taimoorzaeem) [reply test/io/fixtures/roles.sql]: Umm, not sure about this one. This is an edge case so why do we need to test it in all test cases? If we rename our authenticator role to say `"Postgrest_Test_Authenticator"`, it wouldn't be apparent why it is named that way.

Comment (wolfgangwalther) [reply test/io/fixtures/roles.sql]: A comment right above the command to create the role should suffice for explanation, I think.

The advantage of that, in principle, is that you'll be testing a lot more cases that way. What if we add another place of using `current_user` in the future, that is not covered by your custom test? You'd be getting the coverage of *all* our tests that way.

**Review by wolfgangwalther — APPROVED** [review]
Thank you!

**Comment by steve-chavez** [comment]
@taimoorzaeem Should we backport this one?

**Comment by taimoorzaeem** [comment]
> @taimoorzaeem Should we backport this one?

Yes, this should be good to go. It doesn't apply cleanly though, so I will resolve conflicts and do it :).
